### DISCUSSION TITLE
personal_sign request causing error 400 with WalletConnect RPC: "invalid json request"

### DISCUSSION THREAD

**Discussion by BenAzlay** [WalletConnect/walletconnect-monorepo#3407] [Developer Support]
**Describe the bug**
When trying to sign a message with personal_sign, I am not being asked to sign it on my wallet. Rather, the following error is thrown by the WalletConnect RPC endpoint:
```
{"jsonrpc":"2.0","error":{"code":-32600,"message":"invalid json request"}}
```

**To Reproduce**
1. Go to https://app.flashstake.io
2. Click on CONNECT WALLET
3. Connect with WalletConnect
4. Try signing the agreement by clicking "I accept" on the overlay.

**Expected behavior**
Being prompted to sign the message on my wallet when clicking "I accept".

**Desktop (please complete the following information):**
 - OS: Linux
 - Browser: Brave
 - Wallet: Metamask app through WalletConnect V2

**Additional context**
My hypothesis for why this bug happens is that in the WalletConnect documentation, personal_sign is expected to have two params:
1. message to sign
2. address

But in reality, personal_sign has three params, the third being: *password*
The WalletConnect RPC endpoint is not expecting this third param, therefore it throws the error. And the password param will be sent no matter what since even if it's left undefined in the code, like with the code below, the JSON request will have a third null parameter:
```JS
await web3.eth.personal.sign(messageToSign, account)
```

In other words, WalletConnect rather needs to be able to accept requests sent like this:
```JS
await web3.eth.personal.sign(messageToSign, account, '')
```

**Comment by glitch-txs** [comment]
Hi, does this issue happens with other wallets? or only MetaMask?
Just tested MetaMask on android and works fine

**Comment by Abdullah1738** [comment]
Running into the same issue; any resolution?
